packages/mui-material-next/src/Button/Button.tsx

```diff
@@ -542,7 +542,13 @@ const Button = React.forwardRef(function Button<
       as={ComponentProp}
       className={clsx(classes.root, className)}
       ownerState={ownerState}
-      {...getRootProps(getRippleHandlers(props) as unknown as EventHandlers)}
+      {...getRootProps({
```

Comment:
Instead of this, I would propose using the `useSlotProps` hook by providing all the valid parameters. As far as I can see we are not handling the `onClick` event anyhow, it could probably be just passed with the other props.

Reply:
You can check the Joy UI's Button implementation.